src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1852,13 +1837,85 @@
     </Field>
     <Field Name="Value" Type="ExpressionSyntax"/>
   </Node>
+
+  <AbstractNode Name="VariableComponentSyntax" Base="CSharpSyntaxNode">
+  </AbstractNode>
+  <Node Name="TypedVariableComponentSyntax" Base="VariableComponentSyntax">
+    <Kind Name="TypedVariableComponent"/>
+    <Field Name="Type" Type="TypeSyntax"/>
+    <Field Name="Designation" Type="VariableDesignationSyntax"/>
+  </Node>
+  <Node Name="TupleDeconstructionVariableComponentSyntax" Base="VariableComponentSyntax">
+    <Kind Name="TupleDeconstructionVariableComponent"/>
+    <Field Name="OpenParenToken" Type="SyntaxToken">
+      <Kind Name="OpenParenToken"/>
+    </Field>
+    <Field Name="Variables" Type="SeparatedSyntaxList&lt;VariableComponentSyntax&gt;"/>
+    <Field Name="CloseParenToken" Type="SyntaxToken">
+      <Kind Name="CloseParenToken"/>
+    </Field>
+  </Node>
+  <!--
+  <Node Name="WildcardVariableComponentSyntax" Base="VariableComponentSyntax">
+    <Kind Name="WildcardVariableComponent"/>
+    <Field Name="AsteriskToken" Type="SyntaxToken">
+      <Kind Name="AsteriskToken"/>
+    </Field>
+  </Node>
+  -->
+
+  <AbstractNode Name="VariableDesignationSyntax" Base="CSharpSyntaxNode">
+  </AbstractNode>
+  <Node Name="SingleVariableDesignationSyntax" Base="VariableDesignationSyntax">
+    <Kind Name="SingleVariableDesignation"/>
+    <Field Name="Identifier" Type="SyntaxToken">
+      <Kind Name="IdentifierToken"/>
+    </Field>
+  </Node>
+  <Node Name="TupleDeconstructionVariableDesignationSyntax" Base="VariableDesignationSyntax">
+    <Kind Name="TupleDeconstructionVariableDesignation"/>
+    <Field Name="OpenParenToken" Type="SyntaxToken">
+      <Kind Name="OpenParenToken"/>
+    </Field>
+    <Field Name="Variables" Type="SeparatedSyntaxList&lt;VariableDesignationSyntax&gt;"/>
+    <Field Name="CloseParenToken" Type="SyntaxToken">
+      <Kind Name="CloseParenToken"/>
+    </Field>
+  </Node>
+  <!--
+  <Node Name="WildcardVariableDesignationSyntax" Base="VariableDesignationSyntax">
+    <Kind Name="WildcardVariableDesignation"/>
+    <Field Name="AsteriskToken" Type="SyntaxToken">
+      <Kind Name="AsteriskToken"/>
+    </Field>
+  </Node>
+  -->
+
+  <Node Name="DeconstructionDeclarationStatementSyntax" Base="StatementSyntax">
+    <Kind Name="DeconstructionDeclarationStatement"/>
+    <Field Name="Modifiers" Type="SyntaxList&lt;SyntaxToken&gt;"/>
```

Comment:
Yes, if just for good parser error recovery.

---
In reply to: [73615477](https://github.com/dotnet/roslyn/pull/12917#discussion_r73615477) [](ancestors = 73615477)